### PR TITLE
Using 'go install' for default build target to increase compilation speed during development

### PR DESCRIPTION
**Pull request by hferentschik**
Addresses issue #559

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -34,8 +34,8 @@ PACKAGES := go list ./... | grep -v /vendor
 SOURCE_DIRS = cmd pkg test
 DOCS_SYNOPISIS_DIR = ./docs/source/_tmp
 
-$(GOPATH)/bin/minishift$(IS_EXE): $(BUILD_DIR)/$(GOOS)-$(GOARCH)/minishift$(IS_EXE)
-	cp $(BUILD_DIR)/$(GOOS)-$(GOARCH)/minishift$(IS_EXE) $(GOPATH)/bin/minishift$(IS_EXE)
+$(GOPATH)/bin/minishift$(IS_EXE): vendor $(shell $(GOFILES))
+	go install -ldflags="$(LDFLAGS)" ./cmd/minishift
```
Comment (budhrg) [inline Makefile]: @hferentschik Not sure, what this line does. A few lines of brief explanation would help :smile:

Comment (hferentschik) [inline Makefile]: What do you mean, it calls 'go install', right?

**Comment by hferentschik** [comment]
@budhrg I pushed an update to this pull request. I found some other things which were slightly off. Build time is not considerably increased and we are using less Unix specific tools. I wonder whether this not even fixes the default build for Windows when one does not have any cygwin tools (issue #412)

**Comment by budhrg** [comment]
> I wonder whether this not even fixes the default build for Windows when one does not have any cygwin tools (issue #412)

Will verify.

**Comment by hferentschik** [comment]
@budhrg any more feedback? Good to merge? 

@LalatenduMohanty, @gbraad, @praveenkumar - any comments?

**Comment by budhrg** [comment]
LGMT :+1:   Also happen to fix the building of binary in Windows which is issue #412

**Comment by hferentschik** [comment]
Thanks, I'll rebase pull request  #565 now

**Comment by LalatenduMohanty** [comment]
@hferentschik LGTM.

**Pull request closed without merging**